src/OpenFeature.Contrib.Providers.Flagsmith/FlagsmithProvider.cs

```diff
@@ -0,0 +1,167 @@
+        }
+
+        private async Task<ResolutionDetails<T>> ResolveValue<T>(string flagKey, T defaultValue, TryParseDelegate<T> tryParse, EvaluationContext context)
+        {
+
+            var flags = await GetFlags(context);
+            var isFlagEnabled = await flags.IsFeatureEnabled(flagKey);
+            if (!isFlagEnabled)
+            {
+                return new ResolutionDetails<T>(flagKey, defaultValue, reason: Reason.Disabled);
+            }
+
+            var stringValue = await flags.GetFeatureValue(flagKey);
+
+            if (tryParse(stringValue, out var parsedValue))
+            {
+                return new ResolutionDetails<T>(flagKey, parsedValue);
+            }
+            throw new TypeMismatchException("Failed to parse value in the expected type");
+
+        }
+
+
+        /// <inheritdoc/>
+        public override Metadata GetMetadata() => Metadata;
+
+        /// <inheritdoc/>
+
+        public override Task<ResolutionDetails<bool>> ResolveBooleanValue(string flagKey, bool defaultValue, EvaluationContext context = null)
+            => ResolveValue(flagKey, defaultValue, bool.TryParse, context);
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<int>> ResolveIntegerValue(string flagKey, int defaultValue, EvaluationContext context = null)
+            => ResolveValue(flagKey, defaultValue, int.TryParse, context);
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<double>> ResolveDoubleValue(string flagKey, double defaultValue, EvaluationContext context = null)
+            => ResolveValue(flagKey, defaultValue, (string x, out double y) => double.TryParse(x, NumberStyles.Any, CultureInfo.InvariantCulture, out y), context);
+
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<string>> ResolveStringValue(string flagKey, string defaultValue, EvaluationContext context = null)
+            => ResolveValue(flagKey, defaultValue, (string x, out string y) => { y = x; return true; }, context);
+
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<Value>> ResolveStructureValue(string flagKey, Value defaultValue, EvaluationContext context = null)
+            => ResolveValue(flagKey, defaultValue, TryParseValue, context);
+
+        private bool TryParseValue(string stringValue, out Value result)
+        {
+            try
+            {
+                var mappedValue = JsonNode.Parse(stringValue);
+                result = ConvertValue(mappedValue);
+            }
+            catch
+            {
+                result = null;
+            }
+            return result is not null;
+        }
+
+        /// <summary>
+        ///     convertValue is converting the object return by the proxy response in the right type.
+        /// </summary>
+        /// <param name="node">The value we have received</param>
+        /// <returns>A converted object</returns>
```

Comment:
I'm not sure what we mean by `the proxy response` here? Would the following make more sense? 

```suggestion
        /// <summary>
        ///     convertValue is converting the dynamically typed object received from Flagsmith into the correct type
        /// </summary>
        /// <param name="node">The dynamically typed value we received from Flagsmith</param>
        /// <returns>A correctly typed object representing the flag value</returns>
```

Reply:
I just copied some desctiptions and forgot to change :)